util.js

```diff
@@ -40,8 +53,47 @@ function writeDocs(user, repo, readmeData) {
   });
 }
 
+async function writeDocsWithConfig(user, repo, readmeData, config) {
+  try {
+    Object.keys(config).forEach((key) => {
+      mainConfig[`${key}`] = config[key];
+    });
+
+    mainConfig.enablePlugins = false;
+    if (mainConfig.plugins.length > 0) {
+      mainConfig.enablePlugins = true;
+    }
+
+    if (mainConfig.enablePlugins) {
+      const pluginConfig = plugins.addPlugins(mainConfig.plugins);
+      mainConfig.plugins = pluginConfig;
+    }
+
+    const htmlData = docsify.generateHtmlWithConfig(mainConfig);
+
+    fs.writeFileSync(`./docs/${user}-${repo}/README.md`, readmeData, function (
+      err
+    ) {
+      if (err) throw err;
+      console.log("Generated ReadMe!");
+    });
+
+    fs.writeFileSync(`./docs/${user}-${repo}/index.html`, htmlData, function (
+      err
+    ) {
+      if (err) throw err;
+      console.log("Generated HTML!");
+    });
+  } catch (error) {
+    writeDocsWithDefaultConfig(user, repo, readmeData);
+  }
```

Comment:
`try/catch` is not a best practice. 
https://softwareengineering.stackexchange.com/questions/144326/try-catch-in-javascript-isnt-it-a-good-practice

Reply:
Hmm, interesting read. Will look more into error handling with callback functions. Future refactor!!